shuffle/cli/src/shared.rs

```diff
@@ -407,44 +414,59 @@ pub struct NetworksConfig {
 }
 
 impl NetworksConfig {
-    pub fn default() -> Self {
+    pub fn default() -> Result<Self> {
         let mut network_map = BTreeMap::new();
-        network_map.insert("localhost".to_string(), Network::default());
-        NetworksConfig {
+        network_map.insert(LOCALHOST_NAME.to_string(), Network::localhost()?);
+        network_map.insert(TROVE_TESTNET_NAME.to_string(), Network::trove_testnet()?);
+        Ok(NetworksConfig {
             networks: network_map,
-        }
+        })
     }
 
-    pub fn url_for(&self, network_name: &str) -> Result<Url> {
-        let specified_network = self.networks.get(network_name).ok_or_else(|| {
+    pub fn get_network(&self, network_name: &str) -> Result<Network> {
+        Network::from(self.networks.get(network_name).ok_or_else(|| {
             anyhow!("Please add specified network to the ~/.shuffle/Networks.json")
-        })?;
-        Ok(Url::from_str(
-            format!(
-                "{}:{}",
-                specified_network.base, specified_network.dev_api_port
-            )
-            .as_str(),
-        )?)
+        })?)
     }
 }
 
 #[derive(Debug, Serialize, Deserialize, PartialEq)]
 pub struct Network {
     name: String,
-    base: String,
-    json_rpc_port: u16,
-    dev_api_port: u16,
+    json_rpc_url: Url,
+    dev_api_url: Url,
```

Comment:
`dev_api_url` serves for json-rpc as well, so I think we should remove `json_rpc_url`, use `dev_api_url` for json-rpc calls.

Reply:
That might not always be the case, as shown [here](https://fb.quip.com/UytvAaPdWsu4). @avin1100 mentioned that the rust faucet client expects a json rpc url, so there's still some lingering dependencies on it.